tests/metrics/test_intersection.py

```diff
@@ -0,0 +1,48 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from rectools import Columns
+from rectools.metrics.intersection import Intersection
+
+
+class TestIntersection:
+
+    @pytest.mark.parametrize(
+        "k,ref_k,expected_users,expected_intersection",
+        (
+            (2, 2, [1, 2, 4, 5], [0.0, 1.0, 1 / 2, 1 / 3]),
+            (3, None, [1, 2, 3, 4, 5], [1 / 2, 1.0, 1 / 2, 2 / 3, 2 / 4]),
+            (3, 6, [1, 2, 3, 4, 5], [1 / 2, 1.0, 0.0, 1 / 2, 2 / 3]),
+        ),
+    )
+    def test_calc(self, k: int, ref_k: int, expected_users: List[int], expected_intersection: List[float]) -> None:
```

Comment:
We always try to add test on corner cases for metrics. Here I think it is necessary to check that empty `ref_reco` will result in value 0 for the metric.